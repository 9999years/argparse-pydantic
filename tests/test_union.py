from pathlib import Path

import pytest
from pydantic_argparse_extensible import ArgModel
from pytest import CaptureFixture
from syrupy.assertion import SnapshotAssertion


class Args(ArgModel):
    infile: Path | None = None


class Args2(ArgModel):
    infile: None | Path = None


class Args3(ArgModel):
    infile: None | Path | str = None


def test_help(capsys: CaptureFixture, snapshot: SnapshotAssertion) -> None:
    with pytest.raises(SystemExit):
        Args.argparse(args=["--help"])
    assert capsys.readouterr().out == snapshot


def test_help_2(capsys: CaptureFixture, snapshot: SnapshotAssertion) -> None:
    with pytest.raises(SystemExit):
        Args2.argparse(args=["--help"])
    assert capsys.readouterr().out == snapshot


def test_help_3(capsys: CaptureFixture, snapshot: SnapshotAssertion) -> None:
    with pytest.raises(
        TypeError,
        match="Annotation is not callable: None | pathlib.Path | str on field Args3.infile",
    ):
        Args3.argparse(args=[])
    assert capsys.readouterr().out == snapshot


def test_parse() -> None:
    args = Args.argparse(args=[])
    assert args == Args(infile=None)

    args = Args.argparse(args=["--infile", "puppy"])
    assert args == Args(infile=Path("puppy"))
