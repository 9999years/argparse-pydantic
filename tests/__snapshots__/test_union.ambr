# serializer version: 1
# name: test_help
  '''
  usage: pytest [-h] [--infile INFILE]
  
  options:
    -h, --help       show this help message and exit
    --infile INFILE
  
  '''
# ---
# name: test_help_2
  '''
  usage: pytest [-h] [--infile INFILE]
  
  options:
    -h, --help       show this help message and exit
    --infile INFILE
  
  '''
# ---
# name: test_help_3
  ''
# ---
